### ISSUE TITLE
Cannot read properties of undefined (reading 'files')

### ISSUE THREAD

**Issue by sliekens** [microsoft/vscode-remote-release#8647]
Creating a Dev Container from a template throws an ugly error when ghcr.io is unreachable.

![image](https://github.com/microsoft/vscode-remote-release/assets/1583241/17872ece-d6f7-4d45-8d79-023d0807daee)

- VSCode Version: 1.79.2
- Local OS Version: Windows 11 22H2
- Remote OS Version: WSL2 / Debian 11
- Remote Extension/Connection Type: Containers

- Logs:

```
[5373 ms] Dev Containers 0.297.0 in VS Code 1.79.2 (695af097c7bd098fbf017ce3ac85e09bbc5dda06).
[5372 ms]  -- CREATEDEVCONTAINER v2
[11388 ms] Start: Run: C:\Program Files\Microsoft VS Code\Code.exe --ms-enable-electron-run-as-node c:\development\.vscode\extensions\ms-vscode-remote.remote-containers-0.297.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\stelie\AppData\Local\Temp\tmp-output-dir-1687876151825 --template-id ghcr.io/devcontainers/templates/dotnet-mssql:latest --template-args {"imageVariant":"6.0"} --features [] --tmp-dir C:\Users\stelie\AppData\Local\Temp\tmp-dir-1687876151825
[11868 ms] [2023-06-27T14:29:12.318Z] @devcontainers/cli 0.46.0. Node.js v16.17.1. win32 10.0.22621 x64.
[11973 ms] Error: self signed certificate in certificate chain
[11973 ms]     at TLSSocket.onConnectSecure (node:_tls_wrap:1535:34)
[11974 ms]     at TLSSocket.emit (node:events:513:28)
[11975 ms]     at TLSSocket._finishInit (node:_tls_wrap:949:8)
[11975 ms]     at ssl.onhandshakedone (node:_tls_wrap:730:12) {
[11976 ms]   code: 'SELF_SIGNED_CERT_IN_CHAIN'
[11977 ms] }
[11999 ms] Stop (611 ms): Run: C:\Program Files\Microsoft VS Code\Code.exe --ms-enable-electron-run-as-node c:\development\.vscode\extensions\ms-vscode-remote.remote-containers-0.297.0\dist\spec-node\devContainersSpecCLI.js templates apply --workspace-folder C:\Users\stelie\AppData\Local\Temp\tmp-output-dir-1687876151825 --template-id ghcr.io/devcontainers/templates/dotnet-mssql:latest --template-args {"imageVariant":"6.0"} --features [] --tmp-dir C:\Users\stelie\AppData\Local\Temp\tmp-dir-1687876151825
```

Steps to Reproduce:

1. Work behind a corporate proxy/firewall that intercepts TLS traffic to ghcr.io
2. Try to create a new Dev Container from one of the templates

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by sb4tg3o** [comment]
Same issue here, and setting env vars for NODE_EXTRA_CA_CERTS with a path to the cert chain for docker config doesn't resolve this.

**Comment by chrmarti** [comment]
For `NODE_EXTRA_CA_CERTS` there is a fix in Dev Containers 0.297.0-pre-release and later.

**Comment by chrmarti** [comment]
Continuing in https://github.com/microsoft/vscode-remote-release/issues/8623.
